pkg/driver/type/modernbpf.go

```diff
@@ -51,6 +56,31 @@ func (m *modernBpf) HasArtifacts() bool {
 	return false
 }
 
+// Get the private symbol `probeProgram` that will be used to test for
+// type Tracing, attachType AttachTraceRawTp program availability.
+//
+//go:linkname probeProgram github.com/cilium/ebpf/features.probeProgram
+func probeProgram(spec *ebpf.ProgramSpec) error
+
+//nolint:gocritic // the method shall not be able to modify kr
+func (m *modernBpf) Supported(_ kernelrelease.KernelRelease) bool {
```

Comment:
Note: modern bpf will check directly against running kernel, to see if all required features are supported.
This means that we cannot check against "foreign" kernel releases.